inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/convolution/convolution_kernel_base.cpp

```diff
@@ -202,6 +201,10 @@ KernelsData ConvolutionKernelBase::GetCommonKernelsData(const Params& params,
     KernelData kd = KernelData::Default<convolution_params>(params);
     convolution_params& newParams = *static_cast<convolution_params*>(kd.params.get());
 
+    if (!Validate(params, options)) {
+        return {};
```

Comment:
Does it work? `Validate` method checks that
```
bool bSupportedWeightsLayout = params.weights.GetLayout() == GetPreferredWeightsLayout(params);
const bool bWeightsOK = bSupportedWeightsLayout || optParams.allowStaticInputReordering;
```
internally.
So if you call it before `UpdateWeightsParams` then weights layout is not valid yet.

Reply:
You are right, thanks. I moved this checks in `GetCommonKernelsData` from `Validate` but neglected this when pushing changes.